modules/ezcontent_node/modules/ezcontent_smart_article/ezcontent_smart_article.module

```diff
@@ -98,19 +99,28 @@ function ezcontent_smart_article_generate_callback(&$form, FormStateInterface $f
  */
 function ezcontent_smart_summary_generate_callback(&$form, FormStateInterface $form_state) {
   $generate_summary = \Drupal::service('ezcontent_smart_article.generate_summary');
-  $summary_type = $form_state->getValue('field_smart_text_summary')[0]['summary_type'];
-  $number_of_sentences = $form_state->getValue('field_smart_text_summary')[0]['number_of_sentences'];
+  $summary_type = $form_state->getValue('field_smart_text_summary')[0]['summary_container']['summary_type'];
+  $number_of_sentences = $form_state->getValue('field_smart_text_summary')[0]['summary_container']['number_of_sentences'];
   $article = Html::decodeEntities(strip_tags($form_state->getValue('field_smart_text_summary')[0]['value']));
   $text = Json::decode($generate_summary->generateSummary($article, $summary_type, $number_of_sentences));
   $summary = $text['data']['data']['summary'];
 
   $response = new AjaxResponse();
-  $arguments = [$summary, 'edit-field-smart-text-summary-0-summary'];
-  $edit_summary_arguments = ['', '.link-edit-summary'];
+  $arguments = [$summary, 'edit-field-smart-text-summary-0-summary-container-summary-area'];
   $response->addCommand(new InvokeCommand(NULL, "update_summary_text", $arguments));
-  $response->addCommand(new InvokeCommand(NULL, 'update_edit_summary', $edit_summary_arguments));
   return $response;
 }
+/**
+ * Summary generator callback.
+ *
+ * @todo: Move to a service.
+ */
+function update_summary_field_submit_handler(&$form, FormStateInterface $form_state) {
+  $entity = $form_state->getFormObject()->getEntity();
+  $summary = $form_state->getValue('field_smart_text_summary')[0]['summary_container']['summary_area'];
+  $entity->get('field_smart_text_summary')->summary = $summary;
+  $entity->save();
```

Comment:
Do you need to explicitly call Entity->Save ?
Would it not work without it ?

Reply:
I have tested it without Entity->Save  it does not work, we have to save the entity object  after updating it.